sway-lib-std/src/tx.sw

```diff
@@ -205,6 +207,27 @@ pub fn tx_input_coin_owner(input_ptr: u32) -> Address {
     owner_addr
 }
 
+////////////////////////////////////////
+// Inputs > Predicate
+////////////////////////////////////////
+
+pub fn tx_predicate_data_start_offset() -> u64 {
+    let is = instrs_start();
+    let predicate_length_ptr = is - 16;
+    let predicate_code_length = asm(r1, r2: predicate_length_ptr) {
+        lw r1 r2 i0;
+        r1: u64
+    };
+
+    let predicate_data_ptr = is + predicate_code_length;
+    predicate_data_ptr + predicate_data_ptr % 8
```

Comment:
Is this line attempting to round the pointer up to a multiple of 8?  If so, it... isn't.  🙂

`predicate_data_ptr + (8 - (predicate_data_ptr % 8))` is slightly more accurate, except it would add 8 when the ptr is already aligned.

`(predicate_data_ptr + 7) - ((predicate_data_ptr + 7) % 8)` is a better way to do it.

Or if this is not what you were trying to do, adding the value mod 8 to itself doesn't really make sense.

OTOH, if your tests are passing then maybe the predicate data pointer is always aligned anyway.  But I doubt it.

Reply:
Not rounding, but padding. `predicate_data_ptr % 8` is guaranteed to be either `0` (if `predicate_data_ptr` is word-aligned already, i.e. there are an even number of instructions), or `4` (if there are an odd number of instructions).